kratos/geometries/quadrature_point_geometry.h

```diff
@@ -134,87 +136,77 @@ class QuadraturePointGeometry
     {
     }
 
-    explicit QuadraturePointGeometry(const PointsArrayType& ThisPoints)
+    /// Constructor with points, N, Vector<DN_De, ...>
+    QuadraturePointGeometry(
+        const PointsArrayType& ThisPoints,
+        const IntegrationPointType& ThisIntegrationPoint,
+        const Matrix& ThisShapeFunctionsValues,
+        const DenseVector<Matrix>& ThisShapeFunctionsDerivatives)
         : BaseType(ThisPoints, &mGeometryData)
         , mGeometryData(
             &msGeometryDimension,
-            GeometryData::GI_GAUSS_1,
-            {}, {}, {})
+            GeometryShapeFunctionContainerType(
+                GeometryData::GI_GAUSS_1,
+                ThisIntegrationPoint,
+                ThisShapeFunctionsValues,
+                ThisShapeFunctionsDerivatives))
     {
     }
 
-    /**
-     * Copy constructor.
-     * Constructs this geometry as a copy of given geometry.
-     *
-     * @note This copy constructor does not copy the points, thus,
-     * the new geometry shares points with the source geometry.
-     * Any changes to the new geometry points affect the source
-     * geometry points too.
-     */
-    QuadraturePointGeometry( QuadraturePointGeometry const& rOther )
-        : BaseType( rOther )
+    /// Constructor with points, N, Vector<DN_De, ...>, parent
+    QuadraturePointGeometry(
```

Comment:
this constructor seems to be almost the same as the one above
It could be beneficial to call one constructor from the other (check `ModelPart` for how to do this) in order to reduce code duplication

If you want this can be done in a different PR but I think it is worth considering

Reply:
I tried doing that. But it will not work as either I don't assign the parent or I need additional assignments to the foreign constructor. Anyways, thanks for the comment.